File: app/src/main/java/com/nanodegree/popularmovies/AppModule.java
package com.nanodegree.popularmovies;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by luan_ on 31/05/2017.
 */

@Module
public class AppModule {

    Application mApplication;

    public AppModule(Application mApplication) {
        this.mApplication = mApplication;
    }

    @Provides
    @Singleton
    public Application provideApplication() {
        return mApplication;
    }
}

File: app/src/main/java/com/nanodegree/popularmovies/AppModule.kt
package com.nanodegree.popularmovies

import android.app.Application

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Created by luan_ on 31/05/2017.
 */

@Module
class AppModule(internal var mApplication: Application) {

    @Provides
    @Singleton
    fun provideApplicatmion(): Application {
        return mApplication
    }
}
